### PR TITLE
Fix successfully creating directory treated as failure

### PR DESCRIPTION
**Pull request by benblakely**
If the given path does not exist and we’re able to create it then we should continue on.

### CODE REVIEW

File: SWHttpTrafficRecorder.m [hunk 1]
```diff
@@ -75,11 +75,12 @@ - (BOOL)startRecordingAtPath:(NSString *)recordingPath forSessionConfiguration:(
             NSFileManager *fileManager = [NSFileManager defaultManager];
             if(![fileManager fileExistsAtPath:recordingPath]){
                 NSError *bError = nil;
-                [fileManager createDirectoryAtPath:recordingPath withIntermediateDirectories:YES attributes:nil error:&bError];
-                if(bError && error){
-                    *error = [NSError errorWithDomain:SWHttpTrafficRecorderErrorDomain code:SWHttpTrafficRecorderErrorPathFailedToCreate userInfo:@{NSLocalizedDescriptionKey: [NSString stringWithFormat:@"Path '%@' does not exist and error while creating it.", recordingPath]}];
+                if(![fileManager createDirectoryAtPath:recordingPath withIntermediateDirectories:YES attributes:nil error:&bError]){
```
Comment (JinlianWang) [inline SWHttpTrafficRecorder.m:78]: Good catch. Shall we consider the case when bError would not be nil? Currently it is totally ignored.

Comment (benblakely) [inline SWHttpTrafficRecorder.m:78]: @JinlianWang, thanks. When `createDirectoryAtPath:withIntermediateDirectories:attributes:error:` returns `NO`, `bError` will be set and we’ll use it in our error’s `userInfo` dictionary with `NSUnderlyingErrorKey`.

**Comment by JinlianWang** [comment]
Thanks for the explanation, it makes sense.

**Comment by JinlianWang** [comment]
@brblakley Could you help to click the link "Link to CLA" in the Readme.md(or look at the bottom of this page: https://github.com/capitalone/SWHttpTrafficRecorder) and fill out the form there. Take 2 minutes and I am chased after to get this done. Appreciate it!

**Comment by benblakely** [comment]
@JinlianWang OK. I filled out the form and submitted it just now.

**Comment by JinlianWang** [comment]
@brblakley thanks a lot!

**Pull request merged**